### PR TITLE
feat: support `markers: '*'` and custom inline title 

### PR DESCRIPTION
**Pull request by holtwick**
Thanks so much for this great plugin! I was looking for something like this, because I use Obsidian to maintain my posts and was missing support for what they call "callouts". Syntactically they are similar to those of Github alerts. See https://help.obsidian.md/Editing+and+formatting/Callouts

This change allows any "type" of callout and it replaces the title with whatever comes after the `[!xxx]` marker. This is quite useful for multi language content.

You might also like to take a look at collapsable callouts, as an additional feature ;)

### CODE REVIEW

**Comment by antfu** [comment]
Hey, thanks for the PR and reference. I am not so sure we what this relex default behaviour. Maybe optionally allow `makers: string[] | '*'` where `makers: '*'` indicates it's matching everything?

**Comment by holtwick** [comment]
Sure, sounds good for me. I modified the PR accordingly.

File: src/index.ts [hunk 1]
```diff
@@ -54,7 +54,8 @@ const MarkdownItGitHubAlerts: MarkdownIt.PluginWithOptions<MarkdownItGitHubAlert
     classPrefix = 'markdown-alert',
   } = options
 
-  const RE = new RegExp(`^\\[\\!(${markers.join('|')})\\]\\s`, matchCaseSensitive ? '' : 'i')
+  const markerNameRE = markers === '*' ? '\\w+' : markers.join('|')
+  const RE = new RegExp(`^\\[\\!(${markerNameRE})\\]([^\\n\\r]*)`, matchCaseSensitive ? '' : 'i')
```
Comment (antfu) [inline src/index.ts:58]: Can you add a few tests to this custom title feature? Also it would be nice to mention it in the readme

**Comment by jervishs** [comment]
- 非常感谢[antfu](https://github.com/antfu)
- 现在obsidian和vitepress渲染结果得以一致

- 我修改了`/dist/index.mjs`

res

```javascript
const MarkdownItGitHubAlerts = (md, options = {}) => {
  const {
    // markers = ["TIP", "NOTE", "IMPORTANT", "WARNING", "CAUTION",'INFO','DANGER','DETAILS'],
    //更改为仅vitepress支持的block
    markers = ["tip", "warning", 'danger', 'info', 'details'],
    markersList = { tip: "提示", warning: "警告", danger: '危险', info: "信息", details: '详细信息' },
    icons = DEFAULT_GITHUB_ICONS,
    matchCaseSensitive = true,
    titles = {},
    classPrefix = "custom-block"
  } = options;
  //增加*，使标签不存在空格时也进行匹配
  const RE = new RegExp(`^\\[\\!(${markers.join("|")})\\]\\s*`, matchCaseSensitive ? "" : "i");
  md.core.ruler.after("block", "github-alerts", (state) => {
    const tokens = state.tokens;
    for (let i = 0; i < tokens.length; i++) {
      if (tokens[i].type === "blockquote_open") {
        const open = tokens[i];
        const startIndex = i;
        while (tokens[i]?.type !== "blockquote_close" && i <= tokens.length)
          i += 1;
        const close = tokens[i];
        const endIndex = i;
        const firstContent = tokens.slice(startIndex, endIndex + 1).find((token) => token.type === "inline");
        if (!firstContent)
          continue;
        //提前将match转为小写
        const match = firstContent.content.toLowerCase().match(RE);
        if (!match)
          continue;
        const type = match[1];
        const icon = icons[type];
        if (!icon)
          throw new Error(`No icon found for marker ${type}`);
        //获取标题
        const matchTitle = firstContent.content.split('\n')[0];
        //提取标题
        const lastTitle = matchTitle.slice(matchTitle.indexOf(']') + 1).trim();
        //匹配标题
        const title = lastTitle || markersList[type];
        //内容去除标题
        const str = firstContent.content.slice(match[0].length).trimStart();
        firstContent.content = str.replace(title, '').trim();
        open.type = "html_block";
        open.tag = type === 'details' ? "details" : "div";
        if (type === 'details') {
          open.content = `<details class="${classPrefix} ${type}"><summary> ${title ? title : '详细信息'} </summary>`;
          close.content = "</details>";
        } else {
          open.content = `<div class="${classPrefix} ${type}"><p class="${classPrefix}-title flex items-center"><span style='margin-right: 4px;'>${icon}</span>${title}</p>`;
          close.content = "</div>";
        }
        close.tag = type === 'details' ? "details" : "div";
        close.type = "html_block";

      }
    }
  });
};
// function capitalize(str) {
//   return str.charAt(0).toUpperCase() + str.slice(1);
// }

export { MarkdownItGitHubAlerts as default };
```

File: src/index.ts (hunk 1 above)
Comment (holtwick) [reply src/index.ts:58]: Sure, I made these changes to the PR.

**Review by antfu — APPROVED** [review]

**Pull request merged**